sdl_android_lib/src/com/smartdevicelink/transport/SdlPsm.java

```diff
@@ -170,7 +177,11 @@ private int transitionOnInput(byte rawByte, int state){
 				if(dataLength == 0){
 					return FINISHED_STATE; //We are done if we don't have any payload
 				}
-				payload = new byte[dataLength];
+				try {
```

Comment:
Right, but you can actually use the same check as earlier that way we don't have to resort to using the try/catch
```    
    if(dataLength <= V1_V2_MTU_SIZE - V1_HEADER_SIZE){ 
        payload = new byte[dataLength];
    }else{
        return ERROR_STATE;
    }
```